specification/cost-management/resource-manager/Microsoft.CostManagement/preview/2019-03-01-preview/costmanagement.json

```diff
@@ -2966,11 +2940,131 @@
 					"format": "date-time",
 					"type": "string",
 					"readOnly": true
+				}
+			},
+			"allOf": [{
+				"$ref": "#/definitions/ShowbackRulesKind"
+			}]
+		},
+		"CustomPrice": {
+			"x-ms-discriminator-value": "CustomPrice",
+			"type": "object",
+			"description": "Represents Custom price showback rule model",
+			"allOf": [{
+				"$ref": "#/definitions/ShowbackRuleProperties"
+			}],
+			"properties": {
+				"details": {
+					"$ref": "#/definitions/CustomPriceProperties"
+				}
+			}
+		},
+		"CustomPriceProperties": {
+			"type": "object",
+			"title": "Custom price showback rule properties",
+			"properties": {
+				"pricesheet": {
+					"type": "string"
+				},
+				"benefits": {
+					"description": "Array of benefits.",
+					"type": "array",
+					"items": {
+						"type": "string",
+						"enum": [
+							"None", "All", "AHUB", "Reservations"
+						],
+						"x-ms-enum": {
+							"name": "FunctionType",
+							"modelAsString": true
+						}
+					}
 				},
-				"modificationUsername": {
-					"description": "The username who modified the showback rule.",
+				"markups": {
+					"description": "List of markups.",
+					"type": "array",
+					"items": {
+						"$ref": "#/definitions/Markup"
+					}
+				}
+			}
+		},
+		"CustomCharge": {
+			"x-ms-discriminator-value": "CustomCharge",
+			"type": "object",
+			"description": "Represents custom charge showback rule model",
+			"allOf": [{
+				"$ref": "#/definitions/ShowbackRuleProperties"
+			}],
+			"properties": {
+				"details": {
+					"$ref": "#/definitions/CustomChargeProperties"
+				}
+			}
+		},
+		"CustomChargeProperties": {
+			"type": "object",
+			"title": "Custom charge showback rule properties",
+			"properties": {
+				"recurring": {
+					"type": "boolean"
```

Comment:
A string enum with values of "true" and "false" is a more future proof type than boolean. Changing type in the future is a breaking change

Reply:
ok, i changed it to enum